### PR TITLE
TLPORTAL-352: implement live feed for CTools mneme data

### PR DESCRIPTION
**Pull request by dlhaines**
Implement direct http feed for Mneme data from ctools along with
the CTools dashboard feed.  Also fixed some issues with file
retrieval and added additional test files and updated gems and
support tool so could debug from within rubymine.

Implemented in branch [/dlhaines/StudentDashboard/tree/TLPORTAL-352](/dlhaines/StudentDashboard/tree/TLPORTAL-352).

### CODE REVIEW

File: server/data_provider.rb [hunk 1]
```diff
@@ -224,42 +177,142 @@ def dataProviderToDoMnemeLMS(uniqname)
...
-    dataProviderInit
+  ## TODO: should apply to all data sources including ToDo
+  def configureFileProvider(config_hash)
+    dpf_dir = config_hash[:data_provider_file_directory]
+    logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: configure provider file: directory: [#{dpf_dir}]"
 
-    courses = @fileCourses.(uniqname, termid) if @useFileProvider
-    courses = @esbCourses.(uniqname, termid) if @useEsbProvider
+    @useFileProvider = true
+    @fileTerms = Proc.new { |uniqname| dataProviderFileTerms("#{dpf_dir}/terms", uniqname) }
```
Comment (zqian) [inline server/data_provider.rb]: Don't understand why the @fileterms and @fileCourses are dependent on uniqname as shown in the param list.

File: server/data_provider.rb [hunk 2]
```diff
@@ -224,42 +177,142 @@ def dataProviderToDoMnemeLMS(uniqname)
...
-    logIfUnavailable(courses, "courses: user: #{uniqname} term:#{termid}")
+  # Use the ESB to get MPathways information
+  def configureEsbProvider(config_hash)
+    security_file = config_hash[:security_file]
+    application_name = config_hash[:application_name]
+    logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: configure provider esb: security_file: [#{security_file}] application_name: [#{application_name}]"
 
-    courses
+    @useEsbProvider = true
+    @esbTerms = Proc.new { |uniqname| dataProviderESBTerms(uniqname, security_file, application_name) }
```
Comment (zqian) [inline server/data_provider.rb]: same as above - why uniqname is listed as one param here?

File: Gemfile.lock [hunk 3]
```diff
@@ -2,14 +2,15 @@ GEM
   remote: https://rubygems.org/
```
Comment (lsloan) [inline Gemfile.lock]: My Ruby ignorance is probably showing, but I wonder whether what appears to be a lock file should be part of the project repository.

File: server/courselist.rb [hunk 4]
```diff
@@ -87,13 +87,13 @@ class CourseList < Sinatra::Base
   config_hash[:BASE_DIR] = File.dirname(File.dirname(__FILE__))
 
   # name of application to use for ESB information
-  config_hash[:application_name] = "SD-QA"
+  #config_hash[:application_name] = "SD-QA"
```
Comment (lsloan) [inline server/courselist.rb]: Delete the commented out code.

File: Gemfile.lock (hunk 3 above)
Comment (dlhaines) [inline Gemfile.lock]: A fine question.  Including the lock file is a best practice.  Bundler makes it really easy to update versions.  Including the lock file makes it easy to freeze the versions for testing / deployment.

**Pull request merged**